src/service/standard-actions-impl.js

```diff
@@ -205,6 +216,75 @@ export class StandardActions {
     throw user().createError('Unknown AMP action ', method);
   }
 
+  /**
+   * Handles the copy to clipboard action
+   * @param {!./action-impl.ActionInvocation} invocation
+   * @return {!null}
+   */
+  handleCopy_(invocation) {
+    /** @enum {string} */
+    const CopyEvents = {
+      COPY_ERROR: 'copy-error',
+      COPY_SUCCESS: 'copy-success',
+    };
+    let eventName;
+    let eventResult;
+    const {args, node} = invocation;
+    const win = getWin(node);
+
+    let textToCopy;
+    if (invocation.tagOrTarget === 'AMP') {
+      /**
+       * Copy Static Text
+       *  Example: AMP.copy(text='TextToCopy');
+       */
+      textToCopy = args['text'].trim();
+    } else {
+      /**
+       * Copy Target Element Text
+       *  Example: targetId.copy();
+       */
+      const target = devAssertElement(invocation.node);
+      textToCopy = (target.value ?? target.textContent).trim();
+    }
+
+    /**
+     * Trigger Event based on copy action
+     *  - If content got copied to the clipboard successfully, it will
+     *  fire `copy-success` event with data type `success`.
+     *  - If there's any error in copying, it will
+     *  fire `copy-error` event with data type `error`.
+     *  - If browser is not supporting the copy function/action, it
+     *  will fire `copy-error` event with data type `browser`.
+     *
+     *  Example: <button on="tap:AMP.copy(text='Hello AMP');copy-success:copied.show()">Copy</button>
+     */
+    eventName = CopyEvents.COPY_ERROR;
+    if (isCopyingToClipboardSupported(win.document)) {
+      if (copyTextToClipboard(win, textToCopy)) {
+        eventName = CopyEvents.COPY_SUCCESS;
+        eventResult = 'success';
+      } else {
+        eventResult = 'error';
+      }
+    } else {
+      eventResult = 'browser';
+    }
+    const eventValue = /** @type {!JsonObject} */ ({
```

Comment:
Is event value necessary?

Reply:
@jridgewell 

It is necessary but not mandatory. Usually, `eventValue` is useful when when there is an error while copying. It provide three distinct value:
1. `success`: when copying action executed successfully
2. `error`: when there is an error while copying (i.e. end-user disabled copying for web-page, etc.)
3. `browser`: when browser do not support copy functionality

Please let me know your opinion whether to remove this functionality.